src/main/java/com/fasterxml/jackson/core/StreamReadConstraints.java

```diff
@@ -77,8 +77,11 @@ public static final class Builder {
         public Builder maxNestingDepth(final int maxNestingDepth) {
             if (maxNestingDepth < 0) {
                 throw new IllegalArgumentException("Cannot set maxNestingDepth to a negative value");
+            } else if (maxNestingDepth == 0) {
+                this.maxNestingDepth = DEFAULT_MAX_DEPTH;
+            } else {
```

Comment:
-1 from me - behaviour change in a patch release

I think this change is unnecessary

If the javadoc is wrong about the existing code, we can the javadoc.